plugins/workload-management/src/main/java/org/opensearch/plugin/wlm/rule/service/RulePersistenceService.java

```diff
@@ -137,4 +141,32 @@ private void handleGetAllRuleResponse(SearchResponse searchResponse, ActionListe
 
         listener.onResponse(new GetRuleResponse(ruleMap, RestStatus.OK));
     }
+
+    public void deleteRule(String id, ActionListener<DeleteRuleResponse> listener) {
+        if (id == null) {
+            listener.onFailure(new IllegalArgumentException("Rule ID must not be null"));
+            return;
+        }
+
+        // Check if the document exists before deleting
+        GetRequest getRequest = new GetRequest(RULE_INDEX, id);
+        client.get(getRequest, ActionListener.wrap(
+            getResponse -> {
+                if (!getResponse.isExists()) {
+                    listener.onFailure(new ResourceNotFoundException("The rule with id " + id + " doesn't exist"));
+                } else {
+                    // Proceed with deletion
+                    DeleteRequest deleteRequest = new DeleteRequest(RULE_INDEX, id);
+                    client.delete(deleteRequest, ActionListener.wrap(
+                        deleteResponse -> {
+                            boolean acknowledged = deleteResponse.getResult() == DeleteResponse.Result.DELETED;
+                            listener.onResponse(new DeleteRuleResponse(acknowledged));
+                        },
+                        listener::onFailure
+                    ));
+                }
+            },
+            listener::onFailure
+        ));
```

Comment:
What if we directly fire the Delete request would this cause any failures in the execution flow ? I think we should use delete request directly.